packages/plugins/foundry-contracts/ChugSplashLocalExecutor.sol

```diff
@@ -0,0 +1,203 @@
+pragma solidity ^0.8.15;
+
+// SPDX-License-Identifier: MIT
+import { ChugSplashBundles, DeploymentState, BundledChugSplashAction, RawChugSplashAction } from "@chugsplash/contracts/contracts/ChugSplashDataTypes.sol";
+import { UD60x18, ud, convert } from "@prb/math/UD60x18.sol";
+
+abstract contract ChugSplashLocalExecutor {
+    /**
+     * @notice Splits up a bundled action into its components
+     */
+    function disassembleActions(actions[] BundledChugSplashAction) public returns (RawChugSplashAction[], uint256[], bytes32[][]) {
+        RawChugSplashAction[] rawActions;
+        uint256[] memory _actionIndexes;
+        bytes32[][] memory _proofs;
+        for (uint i = 0; i < actions.length; i++) {
+            BundledChugSplashAction action = actions[i];
+            rawActions.push(action.action);
+            _actionIndexes.push(action.proof.actionIndex);
+            _proofs.push(action.proof.siblings);
+        }
+
+        return (rawActions, _actionIndexes, _proofs);
+    }
+
+    /**
+     * Helper function that determines if a given batch is executable within the specified gas limit.
+     */
+    function executable(
+        selected[] BundledChugSplashAction,
+        ChugSplashManager manager,
+        uint masGasLimit
+    ) public view returns (bool) {
+        uint256 initialGasLeft = gasleft();
+
+        (RawChugSplashAction[] actions, uint256[] memory _actionIndexes, bytes32[][] memory _proofs) = disassembleActions(selected);
+
+        manager.executeActions(actions, _actionIndexes, _proofs);
```

Comment:
```suggestion
        (bool success, ) = manager.staticcall{ gas: masGasLimit }(abi.encodeCall(ChugSplashManager.executeActions, (actions, _actionIndexes, _proofs)));
        return success;
```

This mirrors the TypeScript logic. If it looks good to you, we can remove the "gasLeft" stuff in this function

Reply:
Nice that's much better, didn't realize you could do this on chain.